### PR TITLE
Make sure json samples have explicit json ref

### PR DESCRIPTION
**Pull request by andreasohlund**
This also document our reference strategy for samples

### CODE REVIEW

**Comment by andreasohlund** [comment]
@bording turned out that we already have the dependency guidelines documented, can you verify that they are still correct?

https://github.com/Particular/docs.particular.net/blob/master/README.md#nuget-package-references

File: Snippets/ASB/ASB_8/ASB_8.csproj [hunk 1]
```diff
@@ -5,6 +5,7 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="NServiceBus.Azure.Transports.WindowsAzureServiceBus" Version="8.*" />
+    <PackageReference Include="Newtonsoft.Json" Version="11.*" />
```
Comment (bording) [inline Snippets/ASB/ASB_8/ASB_8.csproj]: Snippets don't require having the extra references, just samples.

File: Snippets/ASQ/ASQ_8/ASQ_8.csproj [hunk 2]
```diff
@@ -5,6 +5,7 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="NServiceBus.Azure.Transports.WindowsAzureStorageQueues" Version="8.*" />
+    <PackageReference Include="Newtonsoft.Json" Version="11.*" />
```
Comment (bording) [inline Snippets/ASQ/ASQ_8/ASQ_8.csproj]: Snippets don't require having the extra references, just samples.

File: Snippets/Newtonsoft/Newtonsoft_2/Newtonsoft_2.csproj [hunk 3]
```diff
@@ -5,6 +5,7 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Newtonsoft.Json.Bson" Version="1.*" />
+    <PackageReference Include="Newtonsoft.Json" Version="11.*" />
```
Comment (bording) [inline Snippets/Newtonsoft/Newtonsoft_2/Newtonsoft_2.csproj]: Snippets don't require having the extra references, just samples.

File: Snippets/Rabbit/Rabbit_All/Rabbit_All.csproj [hunk 4]
```diff
@@ -8,7 +8,7 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.*" />
-    <PackageReference Include="Newtonsoft.Json" Version="10.*" />
+    <PackageReference Include="Newtonsoft.Json" Version="11.*" />
```
Comment (bording) [inline Snippets/Rabbit/Rabbit_All/Rabbit_All.csproj]: Snippets don't require having the extra references, just samples.

File: Snippets/Sqs/Sqs_All/Sqs_All.csproj [hunk 5]
```diff
@@ -8,7 +8,7 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="AWSSDK.CloudFormation" Version="3.*" />
-    <PackageReference Include="Newtonsoft.Json" Version="10.*" />
+    <PackageReference Include="Newtonsoft.Json" Version="11.*" />
```
Comment (bording) [inline Snippets/Sqs/Sqs_All/Sqs_All.csproj]: Snippets don't require having the extra references, just samples.

File: Snippets/Templates/Templates_2/Templates_2.csproj [hunk 6]
```diff
@@ -5,6 +5,7 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="NServiceBus" Version="7.*" />
+    <PackageReference Include="Newtonsoft.Json" Version="11.*" />
```
Comment (bording) [inline Snippets/Templates/Templates_2/Templates_2.csproj]: Snippets don't require having the extra references, just samples.

**Comment by andreasohlund** [comment]
SQS snippets fails with

```
  C:\BuildAgent\work\273e95acce0e59c5\Snippets\Sqs\Sqs_All\Sqs_All.csproj : error NU1107: Version conflict detected for Newtonsoft.Json. Reference the package directly from the project to resolve this issue. 
[19:05:12]C:\BuildAgent\work\273e95acce0e59c5\Snippets\Sqs\Sqs_All\Sqs_All.csproj : error NU1107:  Sqs_All -> NServiceBus.Newtonsoft.Json 2.1.0 -> Newtonsoft.Json (>= 11.0.1 && < 12.0.0) 
[19:05:12]C:\BuildAgent\work\273e95acce0e59c5\Snippets\Sqs\Sqs_All\Sqs_All.csproj : error NU1107:  Sqs_All -> NServiceBus.AmazonSQS 4.0.0 -> Newtonsoft.Json (>= 10.0.1 && < 11.0.0).
```

Will add back that explicit ref

**Pull request merged**